lib/ecto/adapters/tds/connection.ex

```diff
@@ -1477,6 +1477,12 @@ if Code.ensure_loaded?(Tds) do
 
     defp quote_table(nil, name), do: quote_table(name)
 
+    defp quote_table({server, db, schema}, name),
+      do: Enum.map_join([quote_table(server), ".", quote_table(db), ".", quote_table(schema), ".", quote_table(name)], "", &"#{&1}")
+
+    defp quote_table({db, schema}, name),
+      do: Enum.map_join([quote_table(db), ".", quote_table(schema), ".", quote_table(name)], "", &"#{&1}")
+
     defp quote_table(prefix, name),
       do: Enum.map_join([quote_table(prefix), ".", quote_table(name)], "", &"#{&1}")
```

Comment:
All of these map joins are returning the structure itself, i don't think you need them:

```suggestion
    defp quote_table({server, db, schema}, name),
      do: [quote_table(server), ".", quote_table(db), ".", quote_table(schema), ".", quote_table(name)]

    defp quote_table({db, schema}, name),
      do: [quote_table(db), ".", quote_table(schema), ".", quote_table(name)]

    defp quote_table(prefix, name),
      do: [quote_table(prefix), ".", quote_table(name)]
```

Also, please add tests to test/ecto/adapters/tds_test.exs. See "delete_all with prefix" as an example, except you don't need to call `delete_all`, `all` is fine. Thanks!